### PR TITLE
BREAKING CHANGE: fastify v5

### PR DESCRIPTION
**Pull request by ilteoood**
- support for fastify v5
- migration to node test runner
 
Due to the structure of the tests I've not been able to run them locally

**Note: this contains a breaking change**

### CODE REVIEW

**Review by simoneb — DISMISSED** [review]

**Comment by simoneb** [comment]
@ilteoood CI is failing, can you take a look?

**Comment by ilteoood** [comment]
> @ilteoood CI is failing, can you take a look?

Sure thing

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,7 +24,6 @@ jobs:
       - name: 'Authenticate with GCP'
         uses: google-github-actions/auth@v2
         with:
-          workload_identity_provider: ${{ vars.GCP_WORKLOAD_IDENTITY_PROVIDER }}
```
Comment (simoneb) [inline .github/workflows/ci.yml:27]: why this change?

Comment (ilteoood) [reply .github/workflows/ci.yml:27]: It was a change I did to see if the CI could work again, but I already reverted it.
GitHub seems to have some problems:
image

**Comment by simoneb** [comment]
@ilteoood github seems to be having some issues with this repo, try maybe force pushing to trigger another sync
